lib/couchdb.js

```diff
@@ -0,0 +1,99 @@
+var existsSync = require('fs').existsSync
+var join = require('path').join
+
+var _ = require('lodash')
+var async = require('async')
+var jsonfile = require('jsonfile')
+var log = require('npmlog')
+var randomstring = require('randomstring')
+var request = require('request')
+
+module.exports = function (config, callback) {
+  if (!config.db.url) {
+    var storePath = join(config.paths.data, 'config.json')
+    var storeExists = existsSync(storePath)
+    var store = storeExists && jsonfile.readFileSync(storePath, {
+      throws: false
+    }) || {}
+    var secret = store.couch_httpd_auth_secret
+
+    if (!secret) {
+      secret = randomstring.generate({
```

Comment:
Currently we're expecting the admin user to be created and to be passed via the url. `!config.db.url` means that it's using pouchdb and that's why I'm generating the secret.